Lecture-6/Task4/l6task4/Application.java

```diff
@@ -0,0 +1,91 @@
+package l6task4;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+public class Application {
+
+    private static SimpleDateFormat formatForDate = new SimpleDateFormat("hh:mm:ss:SSS");
+
+    private static List<Double> getArrayList() {
+        List<Double> arrayList = new ArrayList<>();
+        int counter = 0;
+        while (counter < 10000000) {
+            arrayList.add(Math.random());
+            counter++;
+        }
+        return arrayList;
+    }
+
+    private static List<Double> getLinkedList() {
+        List<Double> linkedList = new LinkedList<>();
+        int counter = 0;
+        while (counter < 10000000) {
+            linkedList.add(Math.random());
+            counter++;
+        }
+        return linkedList;
+    }
+
+    private static void showTimeForAddingElementsToList(List<Double> list) {
```

Comment:
All methods like `showTimeForAddingElementsToList()` are similar except list operation. Refactor them using lambdas:
```java
private static void listTestOperation(List<Double> list, Consumer<List<Double>> operation) {
...
       while (counter < 1000) {
            comsumer.accept(list)
            counter++;
        }
...
}
```
and usage:
```java
Consumer<List<Double>> addOperationConsumer = (list) -> list.add(5000000, Math.random());
listTestOperation(list, addOperationConsumer)
```

Reply:
done